main.c

```diff
@@ -847,6 +853,8 @@ static void dev_removed(struct tcmu_device *dev)
 	rdev->flags |= TCMUR_DEV_FLAG_STOPPING;
 	pthread_mutex_unlock(&rdev->state_lock);
 
+	pthread_cond_destroy(&rdev->lock_cond);
+
```

Comment:
Nice find.

This should be moved to the end of dev_removed with the mutext destroy calls. The lock_cond can still be used a couple lines down in tcmur_stop_device -> tcmu_cancel_lock_thread

Reply:
This has been fixed.